client/app/components/project/package-section.hbs

```diff
@@ -16,41 +16,43 @@
         </ol>
       </div>
       <div class="cell large-6">
-        {{#let (flatten (map-by 'invoices' @packages)) as |invoices|}}
-          {{#if invoices}}
+          {{#if this.invoices}}
             <h4>Invoices</h4>
 
-            {{#each invoices as |invoice|}}
-              <Project::PackageSection::InvoiceInfo
-                @invoice={{invoice}}
-              />
-            {{/each}}
+            <ul>
+              {{#each this.invoices as |invoice|}}
+                <Project::PackageSection::InvoiceInfo
+                  @invoice={{invoice}}
+                />
+              {{/each}}
+            </ul>
 
-            <br/>
-            <br/>
-            <button
-              type="button"
-              class="secondary button"
-              {{on 'click' (fn this.beginPayment)}}
-            >
-              Pay
-            </button>
+            {{#if this.hasPayableInvoices}}
+              <br/>
+              <button
+                type="button"
+                class="secondary button"
+                {{on 'click' (fn this.beginPayment)}}
+              >
+                Pay
+              </button>
+            {{/if}}
 
             <Ui::GenericModal
               @show={{this.isPaying}}
               @closeButtonAction={{fn (mut this.isPaying) false}}
             >
               <h3>Invoices to be paid</h3>
 
-              {{#each invoices as |invoice|}}
+              {{#each this.invoices as |invoice|}}
                 <Project::PackageSection::InvoiceInfo
                   @invoice={{invoice}}
                 />
               {{/each}}
 
               <br/>
               <br/>
-              <h3>Total: ${{reduce (fn this.sum) 0 (map-by 'dcpGrandtotal' invoices)}}</h3>
+              <h3>Total: ${{reduce (fn this.sum) 0 (map-by 'dcpGrandtotal' this.invoices)}}</h3>
```

Comment:
whoa never seen this used before in a template, super cool!

Reply:
@trbmcginnis https://github.com/DockYard/ember-composable-helpers